manager/controlapi/node_test.go

```diff
@@ -731,3 +731,171 @@ func TestUpdateNodeDemote(t *testing.T) {
+	err = ts.Store.Update(func(tx store.Tx) error {
+		n := &api.Network{
+			ID: "net1id",
+			Spec: api.NetworkSpec{
+				Annotations: api.Annotations{
+					Name: "net1name",
+				},
+				Attachable: true,
+			},
+		}
+		return store.CreateNetwork(tx, n)
+	})
+	require.NoError(t, err)
+
+	// create some tasks:
+	err = ts.Store.Update(func(tx store.Tx) error {
+		// 1.) A network attachment on the node we're gonna remove
+		task1 := &api.Task{
+			ID:           "task1",
+			NodeID:       "id2",
+			DesiredState: api.TaskStateRunning,
+			Status: api.TaskStatus{
+				State: api.TaskStateRunning,
+			},
+			Spec: api.TaskSpec{
+				Runtime: &api.TaskSpec_Attachment{
+					Attachment: &api.NetworkAttachmentSpec{
+						ContainerID: "container1",
+					},
+				},
+				Networks: []*api.NetworkAttachmentConfig{
+					{
+						Target:    "net1id",
+						Addresses: []string{}, // just leave this empty, we don't need it
+					},
+				},
+			},
+			// we probably don't care about the rest of the fields.
+		}
+		if err := store.CreateTask(tx, task1); err != nil {
+			return err
+		}
+
+		// 2.) A network attachment on the node we're not going to remove
+		task2 := &api.Task{
+			ID:           "task2",
+			NodeID:       "id1",
+			DesiredState: api.TaskStateRunning,
+			Status: api.TaskStatus{
+				State: api.TaskStateRunning,
+			},
+			Spec: api.TaskSpec{
+				Runtime: &api.TaskSpec_Attachment{
+					Attachment: &api.NetworkAttachmentSpec{
+						ContainerID: "container2",
+					},
+				},
+				Networks: []*api.NetworkAttachmentConfig{
+					{
+						Target:    "net1id",
+						Addresses: []string{}, // just leave this empty, we don't need it
+					},
+				},
+			},
+			// we probably don't care about the rest of the fields.
+		}
+		if err := store.CreateTask(tx, task2); err != nil {
+			return err
+		}
+
+		// 3.) A regular task on the node we're going to remove
+		task3 := &api.Task{
+			ID:           "task3",
+			NodeID:       "id2",
+			DesiredState: api.TaskStateRunning,
+			Status: api.TaskStatus{
+				State: api.TaskStateRunning,
+			},
+			Spec: api.TaskSpec{
+				Runtime: &api.TaskSpec_Container{
+					Container: &api.ContainerSpec{},
+				},
+			},
+		}
+		if err := store.CreateTask(tx, task3); err != nil {
+			return err
+		}
+
+		// 4.) A regular task on the node we're not going to remove
+		task4 := &api.Task{
+			ID:           "task4",
+			NodeID:       "id1",
+			DesiredState: api.TaskStateRunning,
+			Status: api.TaskStatus{
+				State: api.TaskStateRunning,
+			},
+			Spec: api.TaskSpec{
+				Runtime: &api.TaskSpec_Container{
+					Container: &api.ContainerSpec{},
+				},
+			},
+		}
+		return store.CreateTask(tx, task4)
+	})
+	require.NoError(t, err)
+
+	// Now, call the function with our nodeID. make sure it returns no error
+	err = ts.Store.Update(func(tx store.Tx) error {
+		return removeNodeAttachments(tx, "id2")
+	})
+	require.NoError(t, err)
+
+	// Now, make sure only task1, the network-attacahed task on id2, was
```

Comment:
It's probably better to watch and wait for task delete event before verifying that the task was deleted.